test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,115 @@
+    failover_host_set_.hosts_.push_back(
+        makeTestHost(info_, "tcp://127.0.0.1:" + std::to_string(port)));
+    ++port;
+  }
+  // TODO: More than two hosts?
+}
+
+// Initializes random and fixed host sets
+void LoadBalancerFuzzBase::initializeLbComponents(
+    test::common::upstream::LoadBalancerTestCase input) {
+  random_.bytestring_ = constructByteVectorForRandom(input.bytestring_for_random_calls());
+  initializeFixedHostSets(input.num_hosts_in_priority_set(), input.num_hosts_in_failover_set());
+}
+
+// Updating host sets is shared amongst all the load balancer tests. Since logically, we're just
+// setting the mock priority set to have certain values, and all load balancers interface with host
+// sets and their health statuses, this action maps to all load balancers.
+void LoadBalancerFuzzBase::updateHealthFlagsForAHostSet(bool failover_host_set,
+                                                        uint32_t num_healthy_hosts,
+                                                        uint32_t num_degraded_hosts,
+                                                        uint32_t num_excluded_hosts) {
+  MockHostSet& host_set = *priority_set_.getMockHostSet(int(failover_host_set));
+  uint32_t i = 0;
+  for (; i < num_healthy_hosts && i < host_set.hosts_.size(); ++i) {
+    host_set.healthy_hosts_.push_back(host_set.hosts_[i]);
+  }
+  for (; i < (num_healthy_hosts + num_degraded_hosts) && i < host_set.hosts_.size(); ++i) {
+    host_set.degraded_hosts_.push_back(host_set.hosts_[i]);
+  }
+
+  for (; i < (num_healthy_hosts + num_degraded_hosts + num_excluded_hosts) &&
+         i < host_set.hosts_.size();
+       ++i) {
+    host_set.excluded_hosts_.push_back(host_set.hosts_[i]);
+  }
+
+  host_set.runCallbacks({}, {});
+}
+
+void LoadBalancerFuzzBase::prefetch() {
+  // TODO: context, could generate it in proto action
+  lb_->peekAnotherHost(nullptr);
+}
+
+void LoadBalancerFuzzBase::chooseHost() {
+  // TODO: context, could generate it in proto action
+  lb_->chooseHost(nullptr);
+}
+
+void LoadBalancerFuzzBase::replay(
+    const Protobuf::RepeatedPtrField<test::common::upstream::LbAction>& actions) {
+  constexpr auto max_actions = 64;
+  for (int i = 0; i < std::min(max_actions, actions.size()); ++i) {
+    const auto& event = actions.at(i);
+    ENVOY_LOG_MISC(trace, "Action: {}", event.DebugString());
+    switch (event.action_selector_case()) {
+    case test::common::upstream::LbAction::kUpdateHealthFlags: {
+      updateHealthFlagsForAHostSet(event.update_health_flags().failover_host_set(),
+                                   event.update_health_flags().num_degraded_hosts(),
+                                   event.update_health_flags().num_excluded_hosts());
+      break;
+    }
+    case test::common::upstream::LbAction::kPrefetch: {
+      prefetch();
+      break;
+    }
+    case test::common::upstream::LbAction::kChooseHost: {
+      chooseHost();
+      break;
+    }
+    default:
+      break;
+    }
```

Comment:
These braces don't have any effect since you're not declaring anything in the reduced scope. I'd recommend removing them.

Reply:
Deleted. Never thought about this, will keep this in mind from now on!